Lib/glyphsLib/builder/kerning.py

```diff
@@ -20,34 +20,69 @@
 
 def to_ufo_kerning(self):
     for master in self.font.masters:
+        master_id = master.id
+        ufo = self._sources[master_id].font
         kerning_source = master.metricsSource  # Maybe be a linked master
         if kerning_source is None:
             kerning_source = master
-        if kerning_source.id in self.font.kerning:
-            kerning = self.font.kerning[kerning_source.id]
-            _to_ufo_kerning(self, self._sources[master.id].font, kerning)
+        if kerning_source.id in self.font.kerningLTR:
+            kerning = self.font.kerningLTR[kerning_source.id]
+            _to_ufo_kerning(self, ufo, kerning)
+        if kerning_source.id in self.font.kerningRTL:
+            kerning = self.font.kerningRTL[kerning_source.id]
+            _to_ufo_kerning(self, ufo, kerning, "RTL")
 
+        # Prune kerning groups that are not used in kerning rules
```

Comment:
you could have `_to_ufo_kerning` method return the set of used groups, make a union of the LTR and RTL ones and use at the end to prune

Reply:
I think the failing tests have to do with roundtripping, your code doesn't handle that.. not sure how to best unblock this

Reply:
Well, now that we have explicitly named RTL classes, round-tripping should be implementable, shouldn't it?